model/github.py

```diff
@@ -72,23 +82,39 @@ def matches_hostname(self, host_name):
 
     def _optional_attributes(self):
         return (
-            'preferred_protocol',
+            'httpUrl',
             'purpose_labels',
+            'sshUrl',
             'technicalUser',
             'technical_users',
         )
 
     def _required_attributes(self):
         return [
             'apiUrl',
+            'available_protocols',
             'disable_tls_validation',
-            'httpUrl',
-            'sshUrl',
             'webhook_token',
         ]
 
     def validate(self):
         super().validate()
+
+        available_protocols = self.available_protocols()
+        if len(available_protocols) == 0:
```

Comment:
nit: better to check for < 1 (as a general rule. len might be defined to always return `>=0`, but there may be other cases. better to be as restrictive as possible when doing validation checks